spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -173,6 +173,10 @@ public Long snapshotId() {
     return snapshotId;
   }
 
+  public String branch() {
+    return branch;
```

Comment:
This wasn't introduced by this commit, but `branch` should be `final` right?

Reply:
it is effectively final as it's only set once. However, it's not marked as `final` due to the way the different constructors in `SparkTable` are called